cypress/PageObjects/AccountCreationPage.js

```diff
@@ -0,0 +1,73 @@
+export class AccountCreationPage {
+    //Locators
+    getURLtext() {
+        return '#account-creation'
+    }
+    getEmailInput() {
+        return cy.get('#email')
+    }
+    getAddressFristNameInptut() {
+        return cy.get('#firstname')
+    }
+    getAddressLastNameInptut() {
+        return cy.get('#lastname')
+    }
+    getSubmitAccountBtn() {
+        return cy.get('#submitAccount > span')
+    }
+
+    //Actions 
+    setDayOfBirth(dob) {
+        if (dob) {
+            var fields = dob.split('.');
+            var day = fields[2].replace(/^0+/, '')
+            var month = fields[1].replace(/^0+/, '')
+            var year = fields[0]
+            cy.get('#days').select(day)
+            cy.get('#months').select(month)
+            cy.get('#years').select(year)
+        } else {
+            cy.log('dob empty ' + dob)
+        }
+    }
+    fillUserData(userObj) {
+
+        //Fill Obligatory presonal data 
+        cy.get('#customer_firstname').type(userObj.firstName)
+        cy.get('#customer_lastname').type(userObj.lastName)
+        cy.get('#passwd').type(userObj.password)
+        cy.get('#address1').type(userObj.address1)
+        cy.get('#city').type(userObj.city)
+        cy.get('#postcode').type(userObj.zip)
+        cy.get('#id_country').select(userObj.country)
+        cy.get('#phone_mobile').type(userObj.phoneMobile)
+        cy.get('#alias').clear().type(userObj.alias)
+        cy.get('#id_state').select(userObj.state)
+
+        //fill not Obligatory data 
+        this.setDayOfBirth(userObj.dob)
+
+        if (userObj.gender) {
+            var genderId = (userObj.gender === 'male') ? '#id_gender1' : '#id_gender2';
+            cy.get(genderId).click()
+        }
+        if (userObj.newsletter) {
```

Comment:
why do we have 'if' conditionals?

Reply:
This list of IFs handles empty data for not Obligatory fields during registration. For example when we register this user 

```
 {
        "gender": "",
        "firstName": "OnlyObligatoryData",
        "lastName": "NoOptionalData",
        "email": "dm2@test.com",
        "password": "Test1111",
        "dob": "",
        "newsletter": null,
        "specialOffersoptin": null,
        "company": "",
        "address1": "150 main St",
        "address2": "",
        "city": "Taylor",
        "state": "Texas",
        "zip": "73123",
        "country": "United States",
        "other": "",
        "phone": "",
        "phoneMobile": "9876543210",
        "alias": "test alias"
    }
```
since it is not obligatory data, I thought that it is possible for it to have empty values in UserRegistrationData